### PR TITLE
Completed repr() changes

### PR DESCRIPTION
**Pull request by olemb**
* Switched from % syntax to .format().
* Updated the docs to use the new repr() format.
* Fixed a mistake in a docs example. (Import from wrong module.)
* Updated the name of the MIDI Association.
* Removed the `__str__()` methods from MidiTrack and MidiFile.

The code in `MidiFile.__repr__()` and `MidiTrack().__repr__()` is a bit hard to follow but it's OK for now.

### CODE REVIEW

**Comment by olemb** [comment]
@bast Can you review this?

**Comment by bast** [comment]
Test stops with a flake8 complaint:
```
mido/midifiles/midifiles.py:478:80: E501 line too long (84 > 79 characters)
```

**Comment by bast** [comment]
Should I fix that? Will you?

**Comment by olemb** [comment]
It looks like I've already fixed this line in my olemb-fix branch. Let's merge that first and the tests should pass here.

**Comment by bast** [comment]
And a PR for that one is upcoming or already open and I overlooked it?

**Comment by olemb** [comment]
No, just about to put it up. Sorry for the confusion.

**Comment by bast** [comment]
I think this one now needs to be rebased?

**Comment by bast** [comment]
Still same problem.

**Comment by olemb** [comment]
Strange. I'll look into it.

**Comment by olemb** [comment]
Ah, I looked at the wrong line. I see the problem now.

I could wrap the line but I want to do a tiny refactor while I'm at it.

**Comment by olemb** [comment]
Have updated it. Checks are running now.

**Comment by olemb** [comment]
All checks have passed 👍🏻

**Review by bast — APPROVED** [review]
